fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.fineract.portfolio.loanaccount.data.LoanTermVariationsData;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepositoryWrapper;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariationType;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariations;
+import org.apache.fineract.portfolio.loanaccount.exception.LoanNotFoundException;
+import org.apache.fineract.portfolio.loanaccount.rescheduleloan.domain.LoanTermVariationsRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class InterestPauseReadPlatformServiceImpl implements InterestPauseReadPlatformService {
+
+    private final PlatformSecurityContext context;
+    private final LoanTermVariationsRepository loanTermVariationsRepository;
+    private final LoanRepositoryWrapper loanRepositoryWrapper;
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseResponseDto> retrieveInterestPauses(Long loanId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByLoanIdAndTermType(loanId,
+                LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseResponse(variations);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseResponseDto> retrieveInterestPausesByExternalId(String loanExternalId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByExternalLoanIdAndTermType(
+                new ExternalId(loanExternalId), LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseResponse(variations);
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPause(ExternalId loanExternalId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        return processInterestPause(() -> {
+            Long loanId = loanRepositoryWrapper.findIdByExternalId(loanExternalId);
+            if (loanId == null) {
+                throw new LoanNotFoundException("Loan not found with External ID: " + loanExternalId);
+            }
+            return loanRepositoryWrapper.findOneWithNotFoundDetection(loanId);
+        }, startDate, endDate, dateFormat, locale);
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPauseByLoanId(Long loanId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        return processInterestPause(() -> loanRepositoryWrapper.findOneWithNotFoundDetection(loanId), startDate, endDate, dateFormat,
```

Comment:
Write operation should not be in a Read service. Would you mind to extract them?

Reply:
Moved to InterestPauseWritePlatformService